enroll_batch_students.py

```diff
@@ -1,71 +1,142 @@
+locally without config.py file nearby, optional parameters: --host, --port, --course become mandatory.
+Additionally could be set optional parameters --prob and --skills if default values are not appropriate.
 """
+import argparse
 import csv
-from itertools import islice
 import json
 import os
 import sys
 
 import requests
 
-from config import EDXADAPT, DEFAULT_PROBABILITIES, DEFAULT_SKILLS
+try:
+    from config import EDXADAPT
+    REQUIRED_PARAMS = False
+except ImportError:
+    REQUIRED_PARAMS = True
+
+DEFAULT_PROBABILITIES = {'pg': 0.25, 'ps': 0.25, 'pi': 0.1, 'pt': 0.5, 'threshold': 0.99}
+
+DEFAULT_SKILLS = ['center', 'shape', 'spread', 'x axis', 'y axis', 'h to d', 'd to h', 'histogram', 'None']
+
+
+def get_parameters(required=REQUIRED_PARAMS):
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter, description='Enroll students from csv file.'
+    )
+
+    parser.add_argument('csvfile', metavar='file.csv', type=str, help='path/to/csv/file with anonymous student ids.')
+
+    parser.add_argument(
+        '--host',
+        dest='host',
+        type=str,
+        default=EDXADAPT['HOST'] if not required else None,
+        required=required,
+        help='host of the edx-adapt server.'
+    )
+    parser.add_argument(
+        '--port',
+        dest='port',
+        type=int,
+        default=EDXADAPT['PORT'] if not required else None,
+        required=required,
+        help='port of the edx-adapt server.'
+    )
+    parser.add_argument(
+        '--course',
+        dest='course_id',
+        type=str,
+        default=EDXADAPT['COURSE_ID'] if not required else None,
+        required=required,
+        help='course student is enrolled in.'
+    )
+    parser.add_argument(
+        '--prob',
+        dest='probabilities',
+        type=json.loads,
+        default=DEFAULT_PROBABILITIES,
+        help=(
+            'dict with BKT model probabilities student will be enrolled with, default is '
+            '\'{"pg": 0.25, "ps": 0.25, "pi": 0.1, "pt": 0.5, "threshold": 0.99}\''
+        )
+    )
+    parser.add_argument(
+        '--skills',
+        dest='skills',
+        type=str,
+        nargs='+',
+        default=DEFAULT_SKILLS,
+        help=(
+            'sequence with skills student will be enrolled with, default are: '
+            '"center" "shape" "spread" "x axis" "y axis" "h to d" "d to h" "histogram" "None"'
+        )
+    )
+    params = parser.parse_args()
+    return vars(params)
 
 
 def get_students_for_enrollment(path_to_file):
     """
     Parse csv file with student anonymous data and prepare a list of students for enrollment in edx-adapt.
     """
-    if not path_to_file:
-        print "Usage is: python enroll_batch_student.py /path/to/file/with/student_ids"
+    if not os.path.exists(path_to_file):
+        print("File does not exist, please try again")
```

Comment:
It usually more helpful to report back what you've tried to access and failed. So it's easier for script user to recheck the input.